mkShell: make it buildable (#153194)
NixOS/nixpkgs: doc/builders/special/mkshell.section.md

# pkgs.mkShell {#sec-pkgs-mkShell}

`pkgs.mkShell` is a specialized `stdenv.mkDerivation` that removes some
repetition when using it with `nix-shell` (or `nix develop`).

## Usage {#sec-pkgs-mkShell-usage}

Here is a common usage example:

```nix
{ pkgs ? import <nixpkgs> {} }:
pkgs.mkShell {
  packages = [ pkgs.gnumake ];

  inputsFrom = [ pkgs.hello pkgs.gnutar ];

  shellHook = ''
    export DEBUG=1
  '';
}
```

## Attributes

* `name` (default: `nix-shell`). Set the name of the derivation.

> * `packages` (default: `[]`). Add build dependencies to the `nix-shell` environment.

Reviewer: ```suggestion
* `packages` (default: `[]`). Add executable packages to the `nix-shell` environment.
```

Dependencies can be `buildInputs`, but these are `nativeBuildInputs`, which are executable.

> * `inputsFrom` (default: `[]`). Add build dependencies of the dependencies to the `nix-shell` environment.

Reviewer: ```suggestion
* `inputsFrom` (default: `[]`). Add build dependencies of the listed derivations to the `nix-shell` environment.
```

> * `shellHook` (default: `""`). Bash script that is executed by `nix-shell`.

Reviewer: ```suggestion
* `shellHook` (default: `""`). Bash statements that are executed by `nix-shell`.
```
